core/store/models/job_spec.go

```diff
@@ -218,43 +217,74 @@ type InitiatorParams struct {
 	ToBlock    *utils.Big        `json:"toBlock,omitempty" gorm:"type:varchar(255)"`
 	Topics     Topics            `json:"topics,omitempty"`
 
-	RequestData     JSON     `json:"requestData,omitempty" gorm:"type:text"`
-	IdleThreshold   Duration `json:"idleThreshold,omitempty"`
-	Feeds           Feeds    `json:"feeds,omitempty" gorm:"type:text"`
-	Threshold       float32  `json:"threshold,omitempty" gorm:"type:float"`
-	Precision       int32    `json:"precision,omitempty" gorm:"type:smallint"`
-	PollingInterval Duration `json:"pollingInterval,omitempty"`
-}
-
-// defaults represents a default value for an initiator parameter. Value should
-// be the default value, and isZero should be a predicate which returns true if
-// the value of an initiator passed to SetDefaultValues should be overwritten
-// with the default value. See FluxMonitorDefaultInitiatorParams.
-type defaults struct {
-	Value  interface{}
-	isZero func(interface{}) bool
-}
-
-// FluxMonitorDefaultInitiatorParams are the default parameters for Flux
-// Monitor Job Specs.
-var FluxMonitorDefaultInitiatorParams = map[string]defaults{
-	"PollingInterval": {Value: MustMakeDuration(time.Minute),
-		isZero: func(v interface{}) bool { return v.(Duration).IsInstant() },
-	},
-}
-
-// SetDefaultValues returns a InitiatorParams with empty fields set to their
-// default value.
-func (i *InitiatorParams) SetDefaultValues(typ string) {
-	if typ == InitiatorFluxMonitor {
-		r := reflect.ValueOf(i)
-		for name, default_ := range FluxMonitorDefaultInitiatorParams {
-			field := reflect.Indirect(r).FieldByName(name)
-			if default_.isZero(field.Interface()) {
-				field.Set(reflect.ValueOf(default_.Value))
-			}
-		}
+	RequestData JSON            `json:"requestData,omitempty" gorm:"type:text"`
+	Feeds       Feeds           `json:"feeds,omitempty" gorm:"type:text"`
+	Precision   int32           `json:"precision,omitempty" gorm:"type:smallint"`
+	Threshold   float32         `json:"threshold,omitempty"`
+	PollTimer   PollTimerConfig `json:"pollTimer,omitempty" gorm:"type:jsonb"`
+	IdleTimer   IdleTimerConfig `json:"idleTimer,omitempty" gorm:"type:jsonb"`
+}
+
+type PollTimerConfig struct {
+	Disabled bool     `json:"disabled,omitempty"`
+	Period   Duration `json:"period,omitempty"`
+}
+
+// Value is defined so that we can store PollTimerConfig as JSONB, because
+// of an error with GORM where it has trouble with nested structs as JSONB.
+// See https://github.com/jinzhu/gorm/issues/2704
+func (ptc PollTimerConfig) Value() (driver.Value, error) {
+	b, err := json.Marshal(ptc)
+	if err != nil {
+		return nil, err
+	}
+	return b, err
+}
+
+// Scan is defined so that we can read PollTimerConfig as JSONB, because
+// of an error with GORM where it has trouble with nested structs as JSONB.
+// See https://github.com/jinzhu/gorm/issues/2704
```

Comment:
I'm not sure it's an issue with nested structs, I think you may always have to do that. E.g., from the comments on that PR:

> For gorm, implementing driver.Scanner is the only way to mark fields of type struct as normal